### PR TITLE
[CALCITE-3815] Add missing SQL standard aggregate functions: EVERY, SO…

### PR DESCRIPTION
**Pull request by HanumathRao**
…ME, INTERSECTION

This PR contains changes to support EVERY, SOME and INTERSECTION aggregation functions.

### CODE REVIEW

**Comment by chunweilei** [comment]
Please change the commit message to the correct format, namely [CALCITE-xxxx] ....

File: site/_docs/reference.md [hunk 1]
```diff
@@ -1766,12 +1766,15 @@ and `LISTAGG`).
...
 | FUSION(multiset)                   | Returns the multiset union of *multiset* across all input values
+| INTERSECTION(multiset)             | Returns the multiset intersection of *multiset* across all input values
 | APPROX_COUNT_DISTINCT(value [, value ]*)      | Returns the approximate number of distinct values of *value*; the database is allowed to use an approximation but is not required to
 | AVG( [ ALL &#124; DISTINCT ] numeric)         | Returns the average (arithmetic mean) of *numeric* across all input values
 | SUM( [ ALL &#124; DISTINCT ] numeric)         | Returns the sum of *numeric* across all input values
 | MAX( [ ALL &#124; DISTINCT ] value)           | Returns the maximum value of *value* across all input values
 | MIN( [ ALL &#124; DISTINCT ] value)           | Returns the minimum value of *value* across all input values
 | ANY_VALUE( [ ALL &#124; DISTINCT ] value)     | Returns one of the values of *value* across all input values; this is NOT specified in the SQL standard
+| SOME( [ condition ] )                         | Returns true if any expression is true. If no rows are selected, the result is NULL
```
Comment (DonnyZone) [inline site/_docs/reference.md]: > If no rows are selected, the result is NULL
Is this an expected behavior?

**Comment by hsyuan** [comment]
Related PR: https://github.com/apache/calcite/pull/1694

File: core/src/main/codegen/templates/Parser.jj [hunk 2]
```diff
@@ -6405,6 +6423,7 @@ SqlIdentifier ReservedFunctionName() :
     |   <FIRST_VALUE>
     |   <FLOOR>
     |   <FUSION>
+    |   <INTERSECTION>
```
[the remark this answers is not in the document]
Comment (HanumathRao) [reply core/src/main/codegen/templates/Parser.jj]: Thanks @DonnyZone  for the review. I put the comment for this question in JIRA. Please kindly refer to it and let me know  if I am missing anything.

File: site/_docs/reference.md (hunk 1 above)
Comment (HanumathRao) [reply site/_docs/reference.md]: I assumed that the semantics for SOME and EVERY are similar to MIN and MAX respectively. Please let me know if it is misleading, I will remove it.

**Comment by HanumathRao** [comment]
> Please change the commit message to the correct format, namely [CALCITE-xxxx] ....

I changed the commit message. Thanks for pointing it out.

File: core/src/main/java/org/apache/calcite/sql/SqlKind.java [hunk 3]
```diff
@@ -966,6 +966,9 @@
   /** The {@code FUSION} aggregate function. */
   FUSION,
 
+  /** The {@code INTERSECTION} aggregate function. */
+  INTERSECTION,
```
Comment (DonnyZone) [inline core/src/main/java/org/apache/calcite/sql/SqlKind.java]: Please also add `INTERSECTION` in `AGGREGATE` EnumSet [1].

https://github.com/apache/calcite/blob/4c3cef95377ea16efca037012385c40822b4426e/core/src/main/java/org/apache/calcite/sql/SqlKind.java#L1159

**Comment by DonnyZone** [comment]
LGTM, thanks for your work! @HanumathRao, could you please rebase and squash the commits.

File: core/src/main/java/org/apache/calcite/sql/SqlKind.java (hunk 3 above)
Comment (HanumathRao) [reply core/src/main/java/org/apache/calcite/sql/SqlKind.java]: @DonnyZone  Thank you for the suggestion. I added INTESECTION, to the Aggregate enumset. I also noticed that ANY_VALUE is also not added so added it.

**Comment by HanumathRao** [comment]
@DonnyZone  @julianhyde  Thanks for the review. I have squashed all the commits and rebased to the latest.

File: core/src/main/java/org/apache/calcite/sql/SqlKind.java (hunk 3 above)
Comment (DonnyZone) [reply core/src/main/java/org/apache/calcite/sql/SqlKind.java]: That's great! Thanks;)

**Pull request merged**